lib/kubernetes-deploy/kubernetes_resource.rb

```diff
@@ -400,16 +413,34 @@ def validate_timeout_annotation
 
       override = DurationParser.new(timeout_annotation).parse!
       if override <= 0
-        @validation_errors << "#{TIMEOUT_OVERRIDE_ANNOTATION} annotation is invalid: Value must be greater than 0"
+        @validation_errors << "#{annotation_prefix} annotation is invalid: Value must be greater than 0"
       elsif override > 24.hours
-        @validation_errors << "#{TIMEOUT_OVERRIDE_ANNOTATION} annotation is invalid: Value must be less than 24h"
+        @validation_errors << "#{annotation_prefix} annotation is invalid: Value must be less than 24h"
       end
     rescue DurationParser::ParsingError => e
-      @validation_errors << "#{TIMEOUT_OVERRIDE_ANNOTATION} annotation is invalid: #{e}"
+      @validation_errors << "#{annotation_prefix} annotation is invalid: #{e}"
+    end
+
+    def validate_annotation_version
+      annotation_keys = @definition.dig("metadata", "annotations")&.keys
+      annotation_keys&.any? do |annotation|
+        if annotation.include?("kubernetes-deploy.shopify.io")
+          @validation_warnings << "#{annotation} is deprecated: Use 'krane.shopify.io' annotations instead"
+        end
+      end
     end
 
     def timeout_annotation
-      @definition.dig("metadata", "annotations", TIMEOUT_OVERRIDE_ANNOTATION)
+      @definition.dig("metadata", "annotations", TIMEOUT_OVERRIDE_ANNOTATION_DEPRECATED) ||
+        @definition.dig("metadata", "annotations", TIMEOUT_OVERRIDE_ANNOTATION)
+    end
+
+    def annotation_prefix
```

Comment:
I think we should DRY this method and `timeout_annotation` to just take the part after the `/`